qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -232,26 +238,67 @@ def _metadata(self):
         """Override the transpiled circuits method to bring in the inst_map.
 
-        The calibrated schedules are transpiled into the circuits using the instruction
-        schedule map. Since instances of :class:`InstructionScheduleMap` are not serializable
-        they should not be in the transpile options. Only instances of :class:`Calibrations`
-        need to be serialized. Here, we pass the instruction schedule map of the calibrations
-        to the transpiler.
+        The transpilation should do the strict minimum to make the circuits hardware compatible.
+        Indeed, calibration experiments are designed with a specific gate sequence in mind. Any
+        transpiler operation that changes this gate sequence may compromise the validity of the
+        calibration experiment. Sub-classes may override this method to define their own
+        transpilation if need be.
 
         Returns:
             A list of transpiled circuits.
         """
-        transpile_opts = copy.copy(self.transpile_options.__dict__)
-        if "inst_map" in transpile_opts:
-            LOG.warning(
-                "Instruction schedule maps should not be present in calibration "
-                "experiments. Overriding with the inst. map of the calibrations."
-            )
+        transpiled = []
+        for circ in self.circuits():
+            circ = self._map_to_physical_qubits(circ)
+            self._attach_calibrations(circ)
+
+            if self.experiment_options.add_measure_schedules:
+                self._attach_measure_schedules(circ)
+
+            transpiled.append(circ)
+
+        return transpiled
 
-        transpile_opts["inst_map"] = self.calibrations.default_inst_map
-        transpile_opts["initial_layout"] = list(self.physical_qubits)
+    def _map_to_physical_qubits(self, circuit: QuantumCircuit) -> QuantumCircuit:
+        """Map program qubits to physical qubits.
 
-        return transpile(self.circuits(), self.backend, **transpile_opts)
+        Args:
+            circuit: The quantum circuit to map to device qubits.
+
+        Returns:
+            A quantum circuit that has the same number of qubits as the backend and where
+            the physical qubits of the experiment have been properly mapped.
+        """
+        initial_layout = Layout.from_intlist(list(self.physical_qubits), *circuit.qregs)
+
+        return PassManager(
+            [
+                SetLayout(initial_layout),
+                FullAncillaAllocation(CouplingMap(self._backend_data.coupling_map)),
+                EnlargeWithAncilla(),
+                ApplyLayout(),
+            ]
+        ).run(circuit)
+
+    def _attach_measure_schedules(self, circuit: QuantumCircuit):
+        """Attach measurement schedules to the circuit."""
+        for qubit in self.physical_qubits:
+            meas_sched = self._cals.get_schedule("measure", qubit)
+            circuit.add_calibration("measure", (qubit,), meas_sched)
```

Comment:
This is not that simple. You must 
- find measure op node, keep qreg-creg mapping separately
- get measure schedule from calibration
- iterate over all instruction blocks
- once you find an instruction with acquire channel then mutate memory slot index to match with the above mapping

This also allows us to avoid the creg conflict issue in parallel experiment. MemorySlot index can be always 0.

Reply:
Removed it here: [d5a21be](https://github.com/Qiskit/qiskit-experiments/pull/952/commits/d5a21bed96b3f0d6745eb733b503d6f22940590c) for a follow-up PR.

Reply:
Issue here: https://github.com/Qiskit/qiskit-experiments/issues/954